scripts/versions_check.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+set -euo pipefail   # Bash "strict mode"
+
+GO_VERSION=1.18
+NODE_VERSION=16.14.0
+RED_BG=$(tput setab 1)
+WHITE_FG=$(tput setaf 7)
+NORMAL_BG=$(tput sgr0;)
+BOLD=$(tput bold)
+error=false;
+
+echo  " "
+
+function parse_semver() {
+    local token="$1"
+    local major=0
+    local minor=0
+    local patch=0
+
+    if egrep '^[0-9]+\.[0-9]+(\.[0-9]+)?' <<<"${token}" >/dev/null 2>&1 ; then
+        # It has the correct syntax.
+        local n=${token//[!0-9]/ }
+        local a=(${n//\./ })
+    fi
+
+    echo "${a[@]}"
+}
+
+function compare_versions() {
+  local expected=($(parse_semver "$1"))
+  local found=($(parse_semver "$2"))
+  local len=${#expected[@]}
+  for (( i=0; i<len; i++ )) ; do
+    if ! [ "${expected[$i]}" == "${found[$i]}" ]; then
+      echo 1
+    fi
+  done
+}
+
+check_node_version() {
+  if [ -f ~/.nvm/nvm.sh ]; then
+    . ~/.nvm/nvm.sh
+  elif command -v brew; then
+    # https://docs.brew.sh/Manpage#--prefix-formula
+    BREW_PREFIX=$(brew --prefix nvm)
+    if [ -f "${BREW_PREFIX}/nvm.sh" ]; then
+      . "${BREW_PREFIX}"/nvm.sh
+    fi
+  fi
+
+  if ! command -v nvm &> /dev/null ; then
+    echo "WARN: not able to configure nvm"
+    exit 1
+  fi
+
+
+  if ! nvm list "${NODE_VERSION}" &> /dev/null; then
+    echo "${RED_BG}${WHITE_FG}${BOLD}node "${NODE_VERSION}" not installed. Please install it with ${NORMAL_BG}"
+    echo "${RED_BG}${WHITE_FG}nvm install "${NODE_VERSION}"                                ${NORMAL_BG}"
+    echo  ""
+    error=true
+  fi
+}
+
+check_go_version() {
```

Comment:
For Node, I think it's ok to enforce the pinned version for full consistency, given that we are using `NVM` as a version manager, allowing people to have multiple Node versions on their machine.

But in Go's case, we are not (yet) using any version manager, and we would like to allow folks with Go `1.19` and forward to also be able to build Kurtosis, but right now this errors out. Could you add support for this in the function?